components/cdc/src/endpoint.rs

```diff
@@ -819,10 +874,35 @@ impl<T: 'static + RaftStoreRouter<E>, E: KvEngine> Endpoint<T, E> {
         }
     }
 
+    // detect outlier raw regions, schedule deregister for outlier raw regions.
+    fn handle_raw_outlier_regions(
+        &self,
+        raw_resolved_regions: &mut ResolvedRegionVec,
+        min_ts: TimeStamp,
+    ) {
+        if let Some(region) = raw_resolved_regions.get_extreme_outlier(min_ts) {
+            if let Some(delegate) = self.capture_regions.get(&region.region_id) {
+                let observe_id = delegate.handle.id;
+                let deregister = Deregister::Delegate {
+                    region_id: region.region_id,
+                    observe_id,
+                    err: Error::Other(box_err!("raw region dead lock")),
```

Comment:
Could you test if TiCDC recovers automatically when a region is deregister by this error?

Reply:
I trigger the deregister process manually, TiCDC can register the region after deregister and has no effect on resolved_ts. All process is working well.